pkg/healthcheck/healthcheck_output.go

```diff
@@ -48,53 +50,214 @@ func (cr CheckResults) RunChecks(observer CheckObserver) bool {
+					Warning:     true,
+				},
+			}
+		}
+
+		if err == nil {
+			if isatty.IsTerminal(os.Stdout.Fd()) {
+				spin.Suffix = fmt.Sprintf(" Running %s extension check", extension)
+				spin.Color("bold") // this calls spin.Restart()
+			}
+			plugin := exec.Command(path, args...)
+			var stdout, stderr bytes.Buffer
+			plugin.Stdout = &stdout
+			plugin.Stderr = &stderr
+			plugin.Run()
+			extensionResults, err := parseJSONCheckOutput(stdout.Bytes())
+			spin.Stop()
+			if err != nil {
+				command := fmt.Sprintf("%s %s", path, strings.Join(args, " "))
+				if len(stderr.String()) > 0 {
+					err = errors.New(stderr.String())
+				} else {
+					err = fmt.Errorf("invalid extension check output from \"%s\" (JSON object expected):\n%s\n[%s]", command, stdout.String(), err)
+				}
+				results.Results = append(results.Results, CheckResult{
+					Category:    CategoryID(extensionCmd),
+					Description: fmt.Sprintf("Running: %s", command),
+					Err:         err,
+					HintURL:     DefaultHintBaseURL + "extensions",
+				})
+				success = false
+			} else {
+				results.Results = append(results.Results, extensionResults.Results...)
+			}
+		}
+
+		var containsErr bool
+		for _, res := range results.Results {
+			if res.Err != nil {
+				containsErr = true
+				break
+			}
+		}
+
+		if output != JSONOutput && i < len(extensions) {
+			// add a new line to space out each check output
+			fmt.Fprintln(wout)
+			if short && !containsErr {
+				fmt.Fprintln(wout, string(results.Results[0].Category))
+				fmt.Fprintln(wout, strings.Repeat("-", len(string(results.Results[0].Category))))
+			}
+		}
+
+		extensionSuccess := RunChecks(wout, werr, results, output, short)
+		if !extensionSuccess {
+			success = false
+		}
+	}
+
+	return success
+}
+
 // RunChecks runs the checks that are part of hc
-func RunChecks(wout io.Writer, werr io.Writer, hc Runner, output string) bool {
+func RunChecks(wout io.Writer, werr io.Writer, hc Runner, output string, short bool) bool {
 	if output == JSONOutput {
 		return runChecksJSON(wout, werr, hc)
 	}
-	return runChecksTable(wout, hc)
+	return runChecksTable(wout, hc, short)
 }
 
-func runChecksTable(wout io.Writer, hc Runner) bool {
+func runChecksTable(wout io.Writer, hc Runner, short bool) bool {
 	var lastCategory CategoryID
+	var prettyPrintResults CheckObserver
+
 	spin := spinner.New(spinner.CharSets[9], 100*time.Millisecond)
 	spin.Writer = wout
 
-	prettyPrintResults := func(result *CheckResult) {
-		if lastCategory != result.Category {
-			if lastCategory != "" {
-				fmt.Fprintln(wout)
+	if short {
+		prettyPrintResults = func(result *CheckResult) {
+			spin.Stop()
+			if result.Retry {
+				if isatty.IsTerminal(os.Stdout.Fd()) {
+					spin.Suffix = fmt.Sprintf(" %s", result.Err)
+					spin.Color("bold") // this calls spin.Restart()
+				}
+				return
 			}
 
-			fmt.Fprintln(wout, result.Category)
-			fmt.Fprintln(wout, strings.Repeat("-", len(result.Category)))
+			status := okStatus
```

Comment:
I think the logic can be simplified.  We can start by checking if `result.Err == nil` and returning immediately if that is true.  From that point on, we can safely assume that `result.Err != nil` which simplifies some of the following logic.